### PR TITLE
(BKR-890) Surface and provide more visibility into where failures occured

### PR DESCRIPTION
**Pull request by samwoods1**
BKR-890 - Print failure message and stack trace inline when a test failure occurs

### CODE REVIEW

File: lib/beaker/test_suite.rb [hunk 1]
```diff
@@ -141,7 +141,14 @@ def summarize(summary_logger)
       #@param [TestCase] test_case The {TestCase} to examine and print results for
       def print_test_result(test_case)
         test_reported = if test_case.exception
-                          "reported: #{test_case.exception.inspect}"
+                          test_file_trace = ""
```
Comment (kevpl) [inline lib/beaker/test_suite.rb]: this block needs indent fixes. A block should only be indented 2 spaces from its parent code.

**Comment by kevpl** [comment]
@samwoods1 is there any way to split these into smaller more focused PRs? It would be much cleaner if each ticket had its own PR.

For instance, it looks like these commits are pretty great at being their own "logically complete units" of implementation, but when I look closer, it appears that the commit for BKR-556 doesn't have a test included with it. This would be much easier to spot if it was its own PR.

With something as fundamental to beaker as the logger, I would much rather err on the side of over-testing than combining a bunch of work so that it goes through shorter test cycles.

**Comment by samwoods1** [comment]
@kevpl Yeah, I'll split them up... As for the lack of unit test, I didn't find any existing tests for summarizing the results and wasn't quite sure how to go about that.

**Comment by samwoods1** [comment]
@puppetlabs-jenkins retest this please

**Comment by samwoods1** [comment]
@puppetlabs-jenkins retest this please

File: lib/beaker/shared/error_handler.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@ def report_and_raise(logger, e, msg)
       logger.pretty_backtrace(bt).each_line do |line|
         logger.error(line)
       end
-      raise e
+      raise e unless e.is_a? Beaker::DSL::FailTest
```
Comment (kevpl) [inline lib/beaker/shared/error_handler.rb]: @samwoods1 what are the consequences of this line of change? At first look, the rest of the PR seems unrelated to the `report_and_raise` method.

**Comment by kevpl** [comment]
last comment is current

**Comment by samwoods1** [comment]
@puppetlabs-jenkins retest this please

File: lib/beaker/shared/error_handler.rb (hunk 2 above)
Comment (samwoods1) [inline lib/beaker/shared/error_handler.rb]: You're right, I was playing with this, but thought I had removed it.. updated PR.

**Comment by samwoods1** [comment]
@puppetlabs-jenkins retest this please

**Comment by er0ck** [comment]
needs tests

**Comment by samwoods1** [comment]
@er0ck I believe the existing test that was modified covers it.  The behavior changed, and it now checks for the new behavior.  The existing :error test also ensures that I don't break the existing error logic.

**Comment by kevpl** [comment]
@er0ck is there something in particular you're looking to be tested that's not covered?

**Comment by er0ck** [comment]
fair.  i'd hoped this would increase some acceptance testing coverage as well, but upon further inspection i suppose this covers that case well enough.  we can increase logger testing elsewhere.
